core/src/main/java/org/opensearch/sql/analysis/HighlightAnalyzer.java

```diff
@@ -30,12 +30,13 @@ public LogicalPlan analyze(UnresolvedExpression projectItem, AnalysisContext con
 
   @Override
   public LogicalPlan visitAlias(Alias node, AnalysisContext context) {
-    if (!(node.getDelegated() instanceof HighlightFunction)) {
+    UnresolvedExpression delegated = node.getDelegated();
+    if (!(delegated instanceof HighlightFunction)) {
       return null;
     }
 
-    HighlightFunction unresolved = (HighlightFunction) node.getDelegated();
+    HighlightFunction unresolved = (HighlightFunction) delegated;
```

Comment:
I wonder if a better pattern to doing this would be to catch a `RuntimeException` if the casting fails, instead of checking for the instanceof.  That way you don't need to check each time, but instead try and catch for a failure.  
Something like: 
```
HighlightFunction unresolved;
try {
  unresolved = (HighlightFunction) delegated;
 } catch (RuntimeException runtimeException) {
  return null;
}

```

Reply:
Not a bad idea to use this approach, though I do think the readability and verbosity of `instanceof` is nice too. It is also the more commonly found approach in the code base.